### PR TITLE
Cleanup amdllpc.cpp

### PR DESCRIPTION
**Pull request by kuhar**
This simplifies error handling in amdllpc.cpp:main and factors out windows path handling into a helper function.

### CODE REVIEW

**Comment by kuhar** [comment]
I don't have a windows machine and haven't tested the windows-specific code.

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JacobHeAMD** [comment]
ok to test

**Comment by JacobHeAMD** [comment]
test this please

File: llpc/tool/amdllpc.cpp [hunk 1]
```diff
@@ -1497,11 +1497,12 @@ static Result processPipeline(ICompiler *compiler, ArrayRef<std::string> inFiles
 }
 
 #ifdef WIN_OS
-// =====================================================================================================================
-// Finds all filenames which can match input file name
-void findAllMatchFiles(const std::string &inFile,           // [in] Input file name, include wildcard
-                       std::vector<std::string> *pOutFiles) // [out] Output file names which can match input file name
-{
+/// ====================================================================================================================
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1500]: /// -> //
We only use /// for interface headers exposed to driver uses

File: llpc/tool/amdllpc.cpp [hunk 2]
```diff
@@ -1497,11 +1497,12 @@ static Result processPipeline(ICompiler *compiler, ArrayRef<std::string> inFiles
...
+/// Finds all filenames which can match input file name
+///
+/// inFile [in]     : Input file name, including a wildcard.
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1503]: For non-interface files (apart from vkgc.h, llpc.h), the parameter comments:
`// @param [out] paramVar : Parameter comments`
Only add [out] and [in,out] tag. No [in] tag.
You can refer existing codes. I see this issue in another your PR.

File: llpc/tool/amdllpc.cpp [hunk 3]
```diff
@@ -1512,23 +1513,52 @@ void findAllMatchFiles(const std::string &inFile,           // [in] Input file n
   }
 
   // Search first file
-  HANDLE pSearchHandle = FindFirstFileA(inFile.c_str(), &data);
-  if (pSearchHandle == INVALID_HANDLE_VALUE) {
+  HANDLE searchHandle = FindFirstFileA(inFile.c_str(), &data);
+  if (searchHandle == INVALID_HANDLE_VALUE) {
     return;
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1518]: Since you touch this code, single-line statement needn't {}

File: llpc/tool/amdllpc.cpp [hunk 4]
```diff
@@ -1512,23 +1513,52 @@ void findAllMatchFiles(const std::string &inFile,           // [in] Input file n
...
 
   // Copy first file's name
-  pOutFiles->push_back(folderName + data.cFileName);
+  outFiles->push_back(folderName + data.cFileName);
 
   // Copy other file names
-  while (FindNextFileA(pSearchHandle, &data)) {
-    pOutFiles->push_back(folderName + data.cFileName);
+  while (FindNextFileA(searchHandle, &data)) {
+    outFiles->push_back(folderName + data.cFileName);
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1526]: The same as above

File: llpc/tool/amdllpc.cpp [hunk 5]
```diff
@@ -1512,23 +1513,52 @@ void findAllMatchFiles(const std::string &inFile,           // [in] Input file n
...
   }
 
   FindClose(pSearchHandle);
-  return;
 }
 #endif
+
+/// ====================================================================================================================
+/// Expands all input files in a platform-specific way.
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1534]: Comment issue is stated above

File: llpc/tool/amdllpc.cpp [hunk 6]
```diff
@@ -1512,23 +1513,52 @@ void findAllMatchFiles(const std::string &inFile,           // [in] Input file n
...
+///
+/// @param expanded [out] : Returned expanded input filenames.
+/// @returns Result::Success on success, Result::ErrorInvalidValue when expansion fails.
+static Result expandInputFilenames(std::vector<std::string> &expanded) {
```
Comment (amdrexu) [inline llpc/tool/amdllpc.cpp:1538]: expanded -> expandedFilenames
I think it is more clear. Change the parameter name in corresponding comments as well.

File: llpc/tool/amdllpc.cpp (hunk 2 above)
Comment (kuhar) [reply llpc/tool/amdllpc.cpp:1503]: Thanks. I wish there was a clang-format/tidy check for this. I always forget what the format is when I switch between projects.

**Comment by kuhar** [comment]
Addressed Rex's comments.

**Comment by kuhar** [comment]
Seem like a spurious failure?

**Comment by trenouf** [comment]
Yes; two agents passed CTS and the third gave up with a spurious out of memory.

**Comment by amdrexu** [comment]
retest this please

**Comment by amdrexu** [comment]
retest this please

**Comment by amdrexu** [comment]
I think the review is finished. Just the testing is stuck,

**Comment by amdvlk-admin** [comment]
### Test summary for commit 825d71746ab77940b9b0029037b44e02b6e70c68
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 955/955 (100.0%)</li>
<li>Failed: 0/955 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/tool/amdllpc.cpp [hunk 7]
```diff
@@ -1499,36 +1499,63 @@ static Result processPipeline(ICompiler *compiler, ArrayRef<std::string> inFiles
...
 
-  // Copy other file names
-  while (FindNextFileA(pSearchHandle, &data)) {
-    pOutFiles->push_back(folderName + data.cFileName);
-  }
+  // Copy other file names.
+  while (FindNextFileA(searchHandle, &data))
+    outFiles->push_back(folderName + data.cFileName);
 
   FindClose(pSearchHandle);
```
Comment (JaxLinAMD) [inline llpc/tool/amdllpc.cpp:1526]: @kuhar , this line breaks windows build, could  you change pSearchHandle to searchHandle?

Comment (kuhar) [reply llpc/tool/amdllpc.cpp]: Done. Thanks for catching.

**Comment by qiaojbao** [comment]
Avoiding unused variables warnings when using assert() in a Release build.
The line 1685, I got a build warning as error.

[00:07:22][make] ../../../../../../../api/compiler/llpc/tool/amdllpc.cpp:1685:31: error: lambda capture 'result' is not used [-Werror,-Wunused-lambda-capture]
[00:07:22][make]   auto onFailure = [compiler, result] {
[00:07:22][make]                             ~~^~~~~~
[00:07:23][make] 1 error generated.

**Comment by JaxLinAMD** [comment]
> Avoiding unused variables warnings when using assert() in a Release build.
> The line 1685, I got a build warning as error with Clang.
> 
> [00:07:22][make] ../../../../../../../api/compiler/llpc/tool/amdllpc.cpp:1685:31: error: lambda capture 'result' is not used [-Werror,-Wunused-lambda-capture]
> [00:07:22][make] auto onFailure = [compiler, result] {
> [00:07:22][make] ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
> [00:07:23][make] 1 error generated.

@kuhar , could you help take a look? I didn't test with clang when merge it.

**Comment by JaxLinAMD** [comment]
we catch this issue earlier if we have clang build on Docker CI ^_^

**Comment by kuhar** [comment]
@JaxLinAMD opened in #632.

**Pull request merged**